com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/IProvider.java

```diff
@@ -13,16 +13,15 @@
 
 import java.util.Map;
 
-import com.microsoft.java.debug.core.IDebugSession;
-
 public interface IProvider {
     /**
      * Initialize this provider.
-     * @param debugSession
-     *            The associated debug session
+     *
+     * @param debugContext
+     *            The associated debug context
      * @param options
      *            the options
      */
-    default void initialize(IDebugSession debugSession, Map<String, Object> options) {
+    default void initialize(IDebugAdapterContext debugContext, Map<String, Object> options) {
```

Comment:
The initialize method was introduced a month ago. But I didn’t find a complete story of using it properly. Seems like there is no certain chance for initializing a provider. Currently it can be called after launching or attaching. Do we really need to maintain a model inside a provider? Can a provider only consists of static functions?

Reply:
If the provide (HCR provider) need some operation over the virtualmachine/threads/callstacks, it need access the context object.

Reply:
The second parameter `options` is not necessary because debugContext contain all common information.

Reply:
I would leave the IProvider refactor later. Currently, the IProvider/IProviderContext/IDebugAdapterContext is in a chaos. 
I would like to take another iteration of refactor beside this changes.